### ISSUE TITLE
v1.0.4 incorrectly identifies Xeon as 64-bit (migrated from Trac #653)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#1260]
Original ticket http://projects.scipy.org/numpy/ticket/653
Reported 2008-02-04 by trac user doneal24, assigned to atmention:dmcooke.

Platform Gentoo Linux on Intel Xeon, 32-bit

```
biowolf # cat /proc/cpuinfo 
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 15
model           : 2
model name      : Intel(R) Xeon(TM) CPU 2.80GHz
stepping        : 9
cpu MHz         : 2799.930
cache size      : 512 KB
physical id     : 0
siblings        : 2
core id         : 0
cpu cores       : 1
fdiv_bug        : no
hlt_bug         : no
f00f_bug        : no
coma_bug        : no
fpu             : yes
fpu_exception   : yes
cpuid level     : 2
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe cid xtpr
bogomips        : 5572.09
clflush size    : 64
biowolf # python
Python 2.4.4 (#1, Jan 15 2008, 12:27:23) 
[GCC 4.1.2 (Gentoo 4.1.2 p1.0.2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from numpy.distutils.cpuinfo import cpu
>>> cpu.is_64bit()
True
>>>
```

Commenting out lines 282-283 in /usr/lib/python2.4/site-packages/numpy/distutils/cpuinfo.py
    #if self.info[0].get('clflush size','')=='64':
    #   return True
will then correctly return the system type as 32bit but this is not the optimal solution.

Incorrect identification of the system as 64-bit causes scipy to fail in many modules.

**Comment by thouis** [comment]
Attachment in Trac by atmention:cournape, 2008-02-14: [cpuinfo_64bits.patch](http://projects.scipy.org/numpy/attachment/ticket/653/cpuinfo_64bits.patch)

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2008-02-14

Here is a patch which should work everywhere. Instead of using OS dependant logic, I think it is better to use python facilities (platform module is available since 2.3).

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2008-03-08

doneal24,

Could you check this and see if it's fixed in the latest svn or numpy 1.0.5 when it comes out? I applied David's patch and want to know if I can close the ticket.

**Comment by thouis** [comment]
Comment in Trac by atmention:jarrodmillman, 2008-03-11

Fixed in r4852.
